packages/admin/src/table/filterbar/FilterBarDirtyFieldsBadge.tsx

```diff
@@ -0,0 +1,60 @@
+import { createStyles, Theme, Typography, WithStyles, withStyles } from "@material-ui/core";
+import { FieldState } from "final-form";
+import React = require("react");
+import { difference } from "./difference";
+import isEqual = require("lodash.isequal");
+
+export type CometAdminFilterBarDirtyFieldsBadgeClassKeys = "hasValueCount";
+
+export const dirtyFieldsCount = (fieldState?: FieldState<any>) => {
+    if (fieldState?.initial) {
+        const diff = difference<any, Record<string, any>>(fieldState?.value, fieldState?.initial);
+        return Object.keys(diff).length;
+    }
+    return Object.keys(fieldState?.value).length;
+};
+
+export const isEqualFunction = (nextValue?: any, preValue?: any) => {
+    if (preValue === nextValue) {
+        return true;
+    } else {
+        return isEqual(nextValue, preValue);
+    }
+};
+
+const styles = (theme: Theme) =>
+    createStyles<CometAdminFilterBarDirtyFieldsBadgeClassKeys, any>({
+        hasValueCount: {
+            backgroundColor: `${theme.palette.grey[100]}`,
+            textAlign: "center",
+            borderRadius: "4px",
+            height: "20px",
+            width: "17px",
+        },
+    });
+
+interface FilterBarDirtyFieldsBadgeProps {
+    fieldState?: FieldState<any>;
+}
+
+const FilterBarDirtyFieldsBadgeComponent: React.FC<WithStyles<typeof styles, true> & FilterBarDirtyFieldsBadgeProps> = ({ fieldState, classes }) => {
+    if (!isEqualFunction(fieldState?.value, fieldState?.initial)) {
+        return (
+            <div className={classes.hasValueCount}>
+                <Typography variant={"subtitle2"}>
+                    {Array.isArray(fieldState?.value)
+                        ? fieldState?.value.length
+                        : typeof fieldState?.value === "object"
+                        ? dirtyFieldsCount(fieldState)
+                        : 1}
+                </Typography>
+            </div>
+        );
+    } else {
+        return null;
+    }
+};
+
+export const FilterBarDirtyFieldsBadge = withStyles(styles, { name: "CometAdminFilterBarDirtyFieldsBadge", withTheme: false })(
```

Comment:
This component is not used at the moment.
How/Where is it supposed to be used ?

Reply:
The FIlterBarDirtyFieldBadge Component is used in the FilterBarPopOverFormField Component :) it is used a default Badge Component. So you are able to implement your own FieldBadge via Prop :) for example if you need a custome way to calculate your dirtyFields Count :)

Reply:
Oh! I missed this. 👍